docs/bigquery-usage.rst

```diff
@@ -34,31 +34,74 @@ Authorization / Configuration
      >>> client = bigquery.Client()
 
 - Override the credentials inferred from the environment by passing explicit
-  ``credentials`` to one of the alternative classmethods, `from_service_account_json`:
+  ``credentials`` to one of the alternative ``classmethod`` factories,
+  `:meth:gcloud.bigquery.client.Client.from_service_account_json`:
 
   .. doctest::
 
      >>> from gcloud import bigquery
      >>> client = bigquery.Client.from_service_account_json('/path/to/creds.json')
 
-  or `from_service_account_p12`:
+  or `:meth:gcloud.bigquery.client.Client.from_service_account_p12`:
 
   .. doctest::
 
      >>> from gcloud import bigquery
      >>> client = bigquery.Client.from_service_account_p12('/path/to/creds.p12', 'jrandom@example.com')
 
-- Override the project inferred from the environment by passing an explicit
-  ``project`` to the constructor, or to either of the alternative classmethods:
+
+Projects
```

Comment:
> Technically Datastore does the same -- almost everything has "project" as a container ultimately

ISTM that in datastore, project (vs. dataset ID) is not surfaced anywhere in the API / docs (even if the default dataset ID is the same as the project ID). 

> Could we treat BQ Datasets the same way we do with Datastore Datasets or Pubsub Topics? where the client needs to know about a project (which we can try to auto-detect), and then you can create Datasets from that client?

That is the default in this doc:  if you don't provide a project, we attempt to infer it, and raise an exception if we cannot.  The only difference here vs. pubsub is that I'm proposing allowing an explicit `project=None` override which turns that detection off, solely to allow the client to be used to discover project IDs.